src/PowerShellEditorServices/Services/TextDocument/Handlers/TextDocumentHandler.cs

```diff
@@ -52,23 +51,19 @@ class TextDocumentHandler : ITextDocumentSyncHandler
 
         public Task<Unit> Handle(DidChangeTextDocumentParams notification, CancellationToken token)
         {
-            List<ScriptFile> changedFiles = new List<ScriptFile>();
+            ScriptFile changedFile = _workspaceService.GetFile(notification.TextDocument.Uri.ToString());
```

Comment:
Assume the rule is that a `TextDocumentChangeEvent`'s `ContentChanges` will only ever touch the same file?

Maybe worth a comment above the for-loop?

Reply:
[Yep](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_didChange)

Reply:
Actually this is totally obvious from the API we have. Strange that this used to make the opposite assumption